src/main/java/com/cst438/controller/EnrollmentController.java

```diff
@@ -14,29 +18,57 @@
 @CrossOrigin(origins = "http://localhost:3000")
 public class EnrollmentController {
 
+    @Autowired
+    EnrollmentRepository enrollmentRepository;
+
     // instructor downloads student enrollments for a section, ordered by student name
     // user must be instructor for the section
     @GetMapping("/sections/{sectionNo}/enrollments")
+    @PreAuthorize("hasAuthority('SCOPE_ROLE_INSTRUCTOR')")
     public List<EnrollmentDTO> getEnrollments(
             @PathVariable("sectionNo") int sectionNo ) {
 
-        // TODO
-		//  hint: use enrollment repository findEnrollmentsBySectionNoOrderByStudentName method
-        //  remove the following line when done
-        return null;
+        List<Enrollment> enrollmentList = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(sectionNo);
+        List<EnrollmentDTO> dto_list = new ArrayList<EnrollmentDTO>();
+        for (Enrollment enrollment : enrollmentList) {
+            dto_list.add(new EnrollmentDTO(
+                enrollment.getEnrollmentId(),
+                enrollment.getGrade(),
+                enrollment.getStudent().getId(),
+                enrollment.getStudent().getName(),
+                enrollment.getStudent().getEmail(),
+                enrollment.getSection().getCourse().getCourseId(),
+                enrollment.getSection().getCourse().getTitle(),
+                enrollment.getSection().getSecId(),
+                enrollment.getSection().getSectionNo(),
+                enrollment.getSection().getBuilding(),
+                enrollment.getSection().getRoom(),
+                enrollment.getSection().getTimes(),
+                enrollment.getSection().getCourse().getCredits(),
+                enrollment.getSection().getTerm().getYear(),
+                enrollment.getSection().getTerm().getSemester()
+            ));
+        }
+        return dto_list;
     }
 
     // instructor uploads enrollments with the final grades for the section
     // user must be instructor for the section
     @PutMapping("/enrollments")
+    @PreAuthorize("hasAuthority('SCOPE_ROLE_INSTRUCTOR')")
     public void updateEnrollmentGrade(@RequestBody List<EnrollmentDTO> dlist) {
+    
```

Comment:
I think we should pass in the Principal argument here so we can check that the caller of the method is actually the instructor of the section. See ```getAssignments``` method in assignmentController for a similar example.